vm/devices/storage/disk_nvme/nvme_driver/src/queues.rs

```diff
@@ -125,3 +173,42 @@ fn advance(n: u32, l: u32) -> u32 {
         0
     }
 }
+
+#[derive(Protobuf, Clone, Debug)]
+#[mesh(package = "underhill")]
+pub struct SubmissionQueueSavedState {
+    #[mesh(1)]
+    pub sqid: u16,
+    #[mesh(2)]
+    pub head: u32,
+    #[mesh(3)]
+    pub tail: u32,
+    #[mesh(4)]
+    pub committed_tail: u32,
+    #[mesh(5)]
+    pub len: u32,
+    #[mesh(7)]
+    pub base_mem: u64,
```

Comment:
Why do we care if the VA changes across a servicing?

Reply:
Technically we don't. But it is also easier to debug if VAs are same as before. For final merge we can remove it, or maybe keep it for debuggability? I'm fine with either.

Reply:
We absolutely must remove it. We can't rely on the same VA being available in subsequent boots due to ASLR and other random changes to the build.

Reply:
Will do